### PR TITLE
Add Contributing guidelines and update third party notices

### PR DESCRIPTION
**Pull request by carlo-808**
- Adds Contributing guidelines
- Updates Third Party Notices with missing Copyright info for Generic Pool

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+### Feature Requests
+
+Feature requests should be submitted in the [Issue tracker](../../issues), with
+a description of the expected behavior & use case. Before submitting an Issue,
+please search for similar ones in the [closed
+issues](../../issues?q=is%3Aissue+is%3Aclosed+label%3Aenhancement).
+
+### Pull Requests
+
+We can only accept PRs for version v4.0.0 or greater due to open source
```
Comment (nijotz) [inline CONTRIBUTING.md:18]: looks like 4.0.0 will be the next version 👍

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+maintainers of open source projects hosted by New Relic. If you are
+contributing to this project, you're welcome to request access to the
+\#oss-contributors channel in the newrelicusers.slack.com workspace. To request
+access, see https://newrelicusers-signup.herokuapp.com/.
+
+## PR Guidelines
+
+### Version Support
+
+When contributing, keep in mind that New Relic customers (that's you!) are running many different versions of Node, some of them pretty old. Changes that depend on the newest version of Node will probably be rejected, especially if they replace something backwards compatible.
```
Comment (nijotz) [inline CONTRIBUTING.md]: Looks like the text above is line wrapped at 80-100 characters, something like that, and then here it's longer lines. Just thought I'd point it out.

**Review by nijotz — DISMISSED** [review]

File: CONTRIBUTING.md (hunk 2 above)
Comment (carlo-808) [reply CONTRIBUTING.md]: I can update to make it consistent. 👍

File: THIRD_PARTY_NOTICES.md [hunk 3]
```diff
@@ -300,7 +300,7 @@ THE SOFTWARE.
 This product includes source derived from [generic-pool](https://github.com/coopernurse/node-pool) ([v2.5.4](https://github.com/coopernurse/node-pool/tree/v2.5.4)), distributed under the [MIT License](https://github.com/coopernurse/node-pool/blob/v2.5.4/README.md):
 
 ```
-MIT License Copyright (c) <year> <copyright holders>
+Copyright (c) 2010-2016 James Cooper <james@bitmechanic.com>
```
Comment (michaelgoin) [inline THIRD_PARTY_NOTICES.md:303]: This is very likely to just get overwritten. Do we know why it was generated wrong? Can we update the third party manifest?  Also, the blob above still says MIT, does that matter?

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+The MySql instrumentation module includes a suite of functional tests which should be used to
+verify your changes don't break existing functionality.
+
+Functional tests against specific versions of instrumented modules are stored
+in `test/versioned/`. They are written in 
+[node-tap](https://github.com/isaacs/node-tap), and have the extension `.tap.js`.
+
+#### Running Tests
+
+To run the test suite locally, you will need a Docker container running MySQL as setup with the [New Relic Node Agent](https://github.com/newrelic/node-newrelic/blob/main/package.json#L131).
```
Comment (michaelgoin) [inline CONTRIBUTING.md]: I'd prob get more specific about how to get it setup. 

It looks like we have pretty good instructions on the Node agent side, so perhaps saying "you can follow the setup instructions here: https://github.com/newrelic/node-newrelic/blob/main/CONTRIBUTING.md#setup".

File: THIRD_PARTY_NOTICES.md (hunk 3 above)
Comment (carlo-808) [reply THIRD_PARTY_NOTICES.md:303]: Legal approved removing MIT License from the copyright line here.  I'm not sure why it didn't get generated.  I'll take a look at the manifest and try to track that down.

Comment (carlo-808) [reply THIRD_PARTY_NOTICES.md:303]: It looks like the generic pool module does not have an individual License file and has the license information at the bottom of the readme.  The oss cli tool probably did not pick it up then.

Comment (michaelgoin) [reply THIRD_PARTY_NOTICES.md:303]: Interesting. I would double-check with legal the metadata above that is OK.

Unfortunately, we are likely to accidentally revert to the old state in teh future. Need the ability to set metadata in the manifest and tell it to not be updatable automatically.

**Review by michaelgoin — APPROVED** [review]
🐑 🤡

**Pull request merged**